lib/stages/BeamBufferSort.cpp

```diff
@@ -0,0 +1,414 @@
+                            FreqIDBeamMetadata* sub_frame_metadata = (FreqIDBeamMetadata*)&output[dest_start];
+                            // Compute dump data fpga_seq_start and ctime
+                            fpga_seq_start_at_dump = dump_frame_metadata -> fpga_seq_start + in_frame_dump_offset;
+                            ctime_at_dump = dump_frame_metadata -> ctime;
+                            add_nsec(ctime_at_dump, (long)(in_frame_dump_offset * time_resolution_nsec));
+			    //INFO("FPAG_start {:d}, freq {:d} in frame {:d} out frame {:d} RA {:f} dec {:f}", fpga_seq_start_at_dump, ch_id, in_frame_dump_offset, output_buff_offset, dump_frame_metadata-> ra, dump_frame_metadata -> dec);
+                            //INFO("C time int {:d} nsec {:d} frame start {:d} time jump {:d}", ctime_at_dump.tv_sec, ctime_at_dump.tv_nsec, dump_frame_metadata -> ctime.tv_sec, (long)(in_frame_dump_offset * time_resolution_nsec));
+                            fill_freq_meta(sub_frame_metadata, ch_id, fpga_seq_start_at_dump, ctime_at_dump, dump_frame_metadata -> stream_id.id,\
+                                           dump_frame_metadata -> dataset_id, dump_frame_metadata -> beam_number, dump_frame_metadata -> ra, \
+				           dump_frame_metadata -> dec, dump_frame_metadata -> scaling);
+			}
+			
+                        // Copy data to out buffer
+			src_start = FreqIDBeamMeta_size + in_frame_dump_offset * num_pol;
+                        memcpy(&output[dest_start + FreqIDBeamMeta_size + output_buff_offset * num_pol], \
+			       &sort_queue[0][ch_id][src_start], copy_size * num_pol);	
+	            }
+                    
+		}
+		//INFO("Old offset in {:d} out {:d} copy {:d}", in_frame_dump_offset, output_buff_offset, copy_size);
+		in_frame_dump_offset += copy_size;
+		output_buff_offset += copy_size;
+		//INFO("Updated out offset {:d}", output_buff_offset);
+
+	    }
+	    // Update time offset
+	    time_offset = frame_nr - frame0;
+	}
+            
+	// Check if the time offset too old
+	if (time_offset < 0){ // The incoming frame is too old
+            INFO("Drop offset {:d} frame_nr{:d}, frame0 {:d}, freq ID {:d}", time_offset,
+                 frame_nr, frame0, freq_bin);
+            INFO("Frame {:d} comes too late and will be droped.", frame_nr);
+            mark_frame_empty(in_buf, unique_name.c_str(), input_frame_id);
+            input_frame_id++;
+            continue;
+        }
+	// Copy in frame to the queue.
+        //INFO("In queue {:d} {:d}", time_offset, freq_bin);
+	FreqIDBeamMetadata* sub_frame_metadata = (FreqIDBeamMetadata*)&sort_queue[time_offset][freq_bin][0];
+	memcpy(sub_frame_metadata, in_metadata, FreqIDBeamMeta_size);
+	// copy data
+	uint8_t* sub_frame_data = &sort_queue[time_offset][freq_bin][FreqIDBeamMeta_size];
+	memcpy(sub_frame_data, input, in_buf->frame_size);
+        // Debug line
+	FreqIDBeamMetadata* debug_meta = (FreqIDBeamMetadata*)&sort_queue[time_offset][freq_bin][0];
```

Comment:
Isn't `debug_meta` equal to `sub_frame_metadata`? Also, should that `INFO` not be `DEBUG`?